src/types.d.ts

```diff
@@ -114,7 +112,11 @@ export interface GrammarSymbols {
 	[rest]?: Grammar | string | null;
 	[tokenize]?: (code: string, grammar: Grammar, Prism: Prism) => TokenStream;
 }
-export type Grammar = GrammarTokens & GrammarSymbols;
+export interface GrammarSpecial {
+	$insertBefore?: GrammarTokens;
+	$delete: string[];
```

Comment:
```suggestion
	$delete?: string[];
```

Otherwise, we can't do `delete grammar.$delete`.